pkg/execution/state/redis_state/client.go

```diff
@@ -0,0 +1,193 @@
+package redis_state
+
+import (
+	"github.com/oklog/ulid/v2"
+	"github.com/redis/rueidis"
+)
+
+const StateDefaultKey = "estate"
+const QueueDefaultKey = "queue"
+
+type FunctionRunStateClient struct {
```

Comment:
Clients and keys are colocated, so that users are forced into using only keys that work with a specific client.